src/router.rs

```diff
@@ -0,0 +1,86 @@
+#[derive(PartialEq, Clone)]
+pub enum MethodType {
+    POST,
+    GET,
+    PATCH,
+    DELETE,
+    CONNECT,
+    OPTIONS,
+    TRACE,
+    HEAD,
+}
+
+#[derive(Clone)]
+pub struct Endpoint {
+    method: MethodType,
+}
+
+#[derive(Default, Clone)]
+pub struct Route {
+    path: String,
+    endpoint: Endpoint,
+}
+
+#[derive(Default, Clone)]
+pub struct Router {
+    routes: Vec<Route>,
+}
+
+impl Default for Endpoint {
+    fn default() -> Self {
+        Endpoint {
+            method: MethodType::GET,
+        }
+    }
+}
+
+impl Router {
+    pub fn new() -> Self {
+        Router::default()
+    }
+
+    pub fn add_route(mut self, route: Route) -> Self {
+        self.routes.push(route);
+        self
+    }
+
+    pub fn get_handler(&self, method: MethodType, path: String) -> Option<&Route> {
+        for route in &self.routes {
```

Comment:
we'll probably switch to a faster impl (hashmap + fnv?) but this is ok for now.

also this might come in handy :) https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find

Reply:
fnvhashmap exists, that's a good idea.

Reply:
Using a vector here it's a quite common practice for handling multiple routes simultaneously in modern web frameworks.  And for the most cases is works pretty well.
However, using hashmaps maybe will be tricky be some reasons (athough it's a good fit for routes with static paths). 
1. How to handle routes that have `dynamic` parts? For example routes like `/api/v1/users/{id}/`, where the `id` part can be vary
2. How to guarantee to devs that they won't get a wrong route / handler when has a "partial path condidence" on iterations and checks over the hashmap? Let's take a look on the following example:
   ```
   route 1 -> `/api/v1/users/`
   route 2 -> `/api/v1/users/{id}/edit` 
   ```
And then let's say that the user send a request for the route 1, but it can accidentally reach the second route, because we're matching the strings (with the `contains` operation?). In addition to it, the order of insertion mostly relies onto the hashmap functions, which doesn't helps a lot here (because we're doing a direct iteration)
For a vector this issue can be solved manually in code via putting routes in the certain order: details endpoints adds first, then adds for a list.

Reply:
That makes total sense @Relrin , let s go with that ! :)